.github/workflows/mypy.yaml

```diff
@@ -0,0 +1,48 @@
+#
+# GitHub actions Python based test runs.
+#
+# Available VMs:
+# https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
+#
+# Example Python workflow
+# https://docs.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#starting-with-the-python-workflow-template
+name: mypy
+
+on:
+  push:
+    branches: [ trunk ]
+  pull_request:
+    branches: [ trunk ]
+
+jobs:
+  test_run:
```

Comment:
this should be renamed `mypy_test_run` as otherwise the configuration inside GitHub protected branches is confusing.

This might be a bug in GitHub, as instead of `mypy/test_run` the branch protection is visible as just `test_run`.